core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -57,7 +61,22 @@ class MapUnit {
     var cannotEnterOceanTilesUntilAstronomy = false
 
     lateinit var owner: String
+
+    /**
+     * Name key of the unit, used for serialization
+     */
     lateinit var name: String
+
+    /**
+     *  Name of this individual unit, usually resulting from promotion
+     */
+    var instanceName: String? = null
+
+    /**
+     * Name which should be displayed in UI
+     */
+    val displayName: String get() = instanceName?: name
```

Comment:
As part of the coding standard in this project, information which is not saved in the game state should be either transient or converted to a function - please change this to a function.